### PR TITLE
KH2: Promise charm logic

### PR DESCRIPTION
**Pull request by JaredWeakStrike**
Please format your title with what portion of the project this pull request is
targeting and what it's changing.

ex. "MyGame4: implement new game" or "Docs: add new guide for customizing MyGame3"

## What is this fixing or adding?
adding a short circuit for promise charm level logic

## How was this tested?
rolled a seed with the option being true and it short circuited. 
rolled a seed without the option and it didn't short circuit

### CODE REVIEW

**Comment by black-sliver** [comment]
Is this a bug fix? I.e. would it roll unexpected seeds without merging this?

**Comment by JaredWeakStrike** [comment]
> Is this a bug fix? I.e. would it roll unexpected seeds without merging this?

no, level logic is all "fake". Nothing is stopping the player from getting all the level checks when the seed starts but its insanely long so I scaled it using visit locking items. Promise charm path unlocks a room that allows you to level extremely fast and its a bit weird that the logic doesnt consider it.

File: worlds/kh2/Rules.py [hunk 1]
```diff
@@ -83,6 +83,8 @@ def hundred_acre_unlocked(self, state: CollectionState, amount) -> bool:
         return state.has(ItemName.TornPages, self.player, amount)
 
     def level_locking_unlock(self, state: CollectionState, amount):
+        if self.world.options.Promise_Charm:
```
Comment (Silvris) [inline worlds/kh2/Rules.py:86]: I feel like it not checking if the player actually has the Promise Charm could create issues, such as having to grind out to Level 50 without any world unlocks.

Comment (JaredWeakStrike) [reply worlds/kh2/Rules.py:86]: oh you right

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**